File: server/bak/ncurses/label.py

# from server.ncurses. import widget

class Label():
    text = ''

    def __init__(self, text, x, y):
        # super(Widget, self).__init__(x, y, len(text), 1, self)
        self.text = text
        self.x = x
        self.y = y

    def setText(self, text):
        self.text = text


    def draw(self, screen):
        screen.move(self.x, self.y)
        screen.printw(self.text)

File: server/bak/ncurses/label.php
<?php

require_once('ncurses/widget.php');

class Label extends Widget {
    private $text;

    function __construct($text, $x, $y) {
        parent::__construct($x, $y, strlen($text), 1, $this);
        $this->text = $text;
        $this->x = $x;
        $this->y = $y;
    }

    function __destruct() {
    }

    function setText($text) {
        $this->text = $text;
    }


    function draw($screen) {
        $screen->move($this->x, $this->y);
        $screen->printw($this->text);
    }
}

?>